src/components/AddressSearch.js

```diff
@@ -0,0 +1,124 @@
+import _ from 'underscore';
+import React, {useRef, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {View, Text} from 'react-native';
+import {GooglePlacesAutocomplete} from 'react-native-google-places-autocomplete';
+import CONFIG from '../CONFIG';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import styles from '../styles/styles';
+import ExpensiTextInput from './ExpensiTextInput';
+
+const propTypes = {
+    /** The label to display for the field */
+    label: PropTypes.string.isRequired,
+
+    /** The value to set the field to initially */
+    value: PropTypes.string,
+
+    /** A callback function when the value of this field has changed */
+    onChangeText: PropTypes.func.isRequired,
+
+    /** Customize the ExpensiTextInput container */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
+
+    ...withLocalizePropTypes,
+};
+const defaultProps = {
+    value: '',
+    containerStyles: null,
+};
+
+const AddressSearch = (props) => {
+    // Using a hook is the only way to see the value of this component
+    const ref = useRef();
+    useEffect(() => {
+        ref?.current?.setAddressText(props.value);
+    }, []);
+
+    /**
+     * @param {Object} details See https://developers.google.com/maps/documentation/places/web-service/details#PlaceDetailsResponses
+     */
+    const saveLocationDetails = (details) => {
+        if (details.address_components) {
+            // Gather the values from the Google details
+            const streetNumber = _.chain(details.address_components)
+                .find(component => _.contains(component.types, 'street_number'))
+                .get('long_name')
+                .value();
```

Comment:
Can we turn this into a function since most of these are similar?

getAddressComponents('route', 'long_name')

Reply:
Lookin' into that